examples/hacspec-curve25519/src/curve25519.rs

```diff
@@ -0,0 +1,113 @@
+use hacspec_lib::*;
+
+public_nat_mod!(
+    type_name: FieldElement,
+    type_of_canvas: FieldCanvas,
+    bit_size_of_field: 256,
+    modulo_value: "7fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffed"
+);
+public_nat_mod!(
+    type_name: Scalar,
+    type_of_canvas: ScalarCanvas,
+    bit_size_of_field: 256,
+    modulo_value: "8000000000000000000000000000000000000000000000000000000000000000"
+);
+
+bytes!(SerializedPoint, 32);
+bytes!(SerializedScalar, 32);
+
+fn mask_scalar(s: SerializedScalar) -> SerializedScalar {
+    let mut k = s;
+    k[0] = k[0] & U8(248u8);
+    k[31] = k[31] & U8(127u8);
+    k[31] = k[31] | U8(64u8);
+    k
+}
+
+fn decode_scalar(s: SerializedScalar) -> Scalar {
+    let k = mask_scalar(s);
+    Scalar::from_byte_seq_le(k)
+}
+
+fn decode_point(u: SerializedPoint) -> (FieldElement, FieldElement) {
+    let mut u_ = u;
+    u_[31] = u_[31] & U8(127u8);
+    (
+        FieldElement::from_byte_seq_le(u_),
+        FieldElement::from_literal(1u128),
+    )
+}
+
+fn encode_point(p: (FieldElement, FieldElement)) -> SerializedPoint {
+    let (x, y) = p;
+    let b = x * y.inv();
+    SerializedPoint::new().update_start(&b.to_byte_seq_le())
+}
+
+fn point_add_and_double(
+    q: (FieldElement, FieldElement),
```

Comment:
Can we use `type`? If so, we should add the `type Point = (FieldElement, FieldElement);` again.

Reply:
Type used.